### PR TITLE
Add support for using Safety switch pins on IOMCU to control ProfiLEDs via bit banging

### PR DESCRIPTION
**Pull request by bugobliterator**
* CPU time cost is minimal, only upto 7us per call is consumed.
* Flash and Data Cost is as follows:

Before:
```
BUILD SUMMARY
Build directory: /home/sidbh/Programming/ardupilot-chibios/build/iomcu-f103-dshot
Target                   Text (B)  Data (B)  BSS (B)  Total Flash Used (B)  Free Flash (B)  External Flash Used (B)
-------------------------------------------------------------------------------------------------------------------
bin/iofirmware_lowpolh      52404      1196    19300                 53600            7836  Not Applicable
bin/iofirmware_highpolh     52404      1196    19300                 53600            7836  Not Applicable
```

After
```
BUILD SUMMARY
Build directory: /home/sidbh/Programming/ardupilot-chibios/build/iomcu-f103-dshot
Target                   Text (B)  Data (B)  BSS (B)  Total Flash Used (B)  Free Flash (B)  External Flash Used (B)

-------------------------------------------------------------------------------------------------------------------
bin/iofirmware_lowpolh      52160      1196    19296                 53356            8084  Not Applicable
bin/iofirmware_highpolh     52160      1196    19296                 53356            8084  Not Applicable
```

### CODE REVIEW

File: libraries/AP_BoardConfig/AP_BoardConfig.cpp [hunk 1]
```diff
@@ -157,7 +157,7 @@ const AP_Param::GroupInfo AP_BoardConfig::var_info[] = {
     // @Param: SAFETY_DEFLT
     // @DisplayName: Sets default state of the safety switch
     // @Description: This controls the default state of the safety switch at startup. When set to 1 the safety switch will start in the safe state (flashing) at boot. When set to zero the safety switch will start in the unsafe state (solid) at startup. Note that if a safety switch is fitted the user can still control the safety state after startup using the switch. The safety state can also be controlled in software using a MAVLink message.
-    // @Values: 0:Disabled,1:Enabled
+    // @Values: 0:Disabled,1:Enabled,3:Use Safety pin to control ProfiLED
```
Comment (tridge) [inline libraries/AP_BoardConfig/AP_BoardConfig.cpp:160]: we still need BRD_SAFETY_DEFLT, most vehicles these days have safety state controlled via mavlink in software, not via a button

**Review by tridge — CHANGES_REQUESTED** [review]
we can't lose the BRD_SAFETY_DFLT feature, needed by many users

File: libraries/AP_BoardConfig/AP_BoardConfig.cpp (hunk 1 above)
Comment (bugobliterator) [reply libraries/AP_BoardConfig/AP_BoardConfig.cpp]: Fixed by creating a separate parameter that enabled IO Profiled control. Also tested if Safety switch control continue to operate as before through mission planner.

**Review by tridge — CHANGES_REQUESTED** [review]
we've discussed, and think this should be a iomcu_CubePilot build, and not make any fw change on other variants
also needs some dshot testing by @andyp1per

**Comment by tridge** [comment]
for the ccache failure, I think we need to change it to be two other H7 boards we test, maybe Pixhawk6X and ZeroOneX6 ?

**Comment by tridge** [comment]
hmm, the ccache test (see test_ccache.yml) uses Durandal-bdshot and Pixhawk6X, which should be unaffected by this PR, which implies this PR is changing something it shouldn't

File: libraries/AP_Notify/AP_Notify.cpp [hunk 2]
```diff
@@ -115,7 +116,11 @@ AP_Notify *AP_Notify::_singleton;
 #endif  // defined (DEFAULT_NTF_LED_TYPES)
 
 #ifndef DEFAULT_NTF_LED_TYPES
-    #define DEFAULT_NTF_LED_TYPES (Notify_LED_Board | I2C_LEDS)
+    #if HAL_WITH_IO_MCU && AP_IOMCU_PROFILED_SUPPORT_ENABLED
+      #define DEFAULT_NTF_LED_TYPES (Notify_LED_Board | Notify_LED_ProfiLED_IOMCU | I2C_LEDS)
+    #else
+        #define DEFAULT_NTF_LED_TYPES (Notify_LED_Board | I2C_LEDS)
+    #endif
```
Comment (peterbarker) [inline libraries/AP_Notify/AP_Notify.cpp:123]:
```suggestion
    #define DEFAULT_NTF_LED_TYPES (Notify_LED_Board | I2C_LEDS)

    #if HAL_WITH_IO_MCU && AP_IOMCU_PROFILED_SUPPORT_ENABLED
    #define DEFAULT_NTF_LED_TYPES (DEFAULT_NTF_LED_TYPES | Notify_LED_ProfiLED_IOMCU)
    #endif
```

**Review by peterbarker — DISMISSED** [review]
Pixhawk6X is also diverging from Durandal somehow, causing ccache to test fail.  That needs to be explained as this shouldn't really affect either board.

**Comment by peterbarker** [comment]
Ping @bugobliterator - those plane trips have to be good for something? :-)

File: libraries/AP_Notify/AP_Notify.cpp (hunk 2 above)
Comment (bugobliterator) [reply libraries/AP_Notify/AP_Notify.cpp:123]: We cannot redefine macros

**Review by andyp1per — APPROVED** [review]
Testing seemed fine

**Comment by peterbarker** [comment]
```
Board                    AP_Periph  blimp  bootloader  copter  heli  iofirmware  plane  rover  sub
CubeBlack                           776    *           776     776               776    776    776
CubeOrange-periph-heavy  408                                                                   
CubeOrangePlus                      784    *           784     776               784    776    776
CubeRedPrimary                      *      *           *       *                 *      *      *
Durandal                            *      *           *       *                 *      *      *
Hitec-Airspeed           *                                                                     
KakuteH7-bdshot                     *      *           *       *                 *      *      *
MatekF405                           *      *           *       *                 *      *      *
Pixhawk1-1M-bdshot                  *                  *       *                 *      *      *
f103-QiotekPeriph        *                                                                     
f303-Universal           *                                                                     
iomcu                                                                *                         
revo-mini                           *      *           *       *                 *      *      *
skyviper-journey                                       *                                       
skyviper-v2450                                         *                                       
```

**Review by tridge — APPROVED** [review]

**Comment by tridge** [comment]
i'd say that using the safety pin like this offends my sensibilities a bit, but holding my nose I will merge

**Pull request merged**